src/blueapi/cli/scratch.py

```diff
@@ -128,3 +129,30 @@ def _validate_directory(path: Path) -> None:
         raise KeyError(f"{path}: No such file or directory")
     elif path.is_file():
         raise KeyError(f"{path}: Is a file, not a directory")
+
+
+def get_scratch_info(config: ScratchConfig) -> ScratchResponse:
+    scratch_responses = ScratchResponse()
+    try:
+        _validate_directory(config.root)
+    except Exception as e:
+        logging.error(f"Failed to get scratch info: {e}")
+        return scratch_responses
+    for repo in config.repositories:
+        local_directory = config.root / repo.name
+        repo = Repo(local_directory)
+        try:
+            branch = repo.active_branch.name
+        except TypeError:
+            branch = repo.head.commit.hexsha
+
+        is_dirty = repo.is_dirty()
+        scratch_responses.package_info.append(
+            RepositoryStatus(
+                repository_name=repo.remotes.origin.url,
```

Comment:
repository name and origin remote url aren't quite synonymous, is this the right name?

What if the repository does not have a remote named origin? It's possible to delete remotes, even origin, for example if you clone with https and then need to add the ssh endpoint as another remote (as I have done too often...)

What does `repo.description` give you, or `repo.remotes[0]`? What if there are multiple remotes?